app/src/androidTest/java/org/schabi/newpipe/testUtil/DatabaseMocker.kt

```diff
@@ -0,0 +1,32 @@
+package org.schabi.newpipe.testUtil
+
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import org.junit.Assert.assertSame
+import org.schabi.newpipe.NewPipeDatabase
+import org.schabi.newpipe.database.AppDatabase
+
+class DatabaseMocker {
```

Comment:
I find the naming misleading since it's not mocking the database but creating an actual in-memory database.

For me a mock is something where by default the methods return "nothing" or the closest thing to nothing (empty objects) unless one specifically defines what should be returned. This does not apply to the in-memory database.

Reply:
Well, it is mocking NewPipeDatabase.databaseInstance with the in-memory database... How would you call it?

Reply:
I think you have a different understanding from me if what a mock is.
I find this explanation decent https://stackoverflow.com/a/17810004

---

Going with that thread it would be a fake. But calling it `TestDatabase` would be fine too

Reply:
A mock would be like in `ContentSettingsManagerTest`
https://github.com/TeamNewPipe/NewPipe/blob/ad405d9e0b63ac96949953f9fca0c125ee35f3c4/app/src/test/java/org/schabi/newpipe/settings/ContentSettingsManagerTest.kt#L56

Reply:
Done